### PR TITLE
Fix last linting errors & enable linting in Travis

### PR DESCRIPTION
**Pull request by ericholscher**
This enables Linting in Travis,
along with fixing the last of the errors.

### CODE REVIEW

File: readthedocs/core/models.py [hunk 1]
```diff
@@ -27,10 +28,6 @@ def __unicode__(self):
         return (ugettext("%(username)s's profile")
                 % {'username': self.user.username})
 
-    def get_form(self):
```
Comment (gregmuellegger) [inline readthedocs/core/models.py]: The `get_form` *is* used as far as I can tell: https://github.com/rtfd/readthedocs.org/blob/master/readthedocs/templates/profiles/public/profile_detail.html#L11
However, the `form` variable is then again not used in this template, and not in the included `project_list_detailed.html` :) So we only need to remove the usage of `get_form` from the template and should be set.

**Comment by gregmuellegger** [comment]
+3 on this :+1: :+1: :+1: :grinning:

File: readthedocs/core/models.py (hunk 1 above)
Comment (ericholscher) [inline readthedocs/core/models.py]: Aye. I grepped the *.py files, not the html also, silly me. Should be fixed now.

File: readthedocs/templates/profiles/public/profile_detail.html [hunk 2]
```diff
@@ -8,7 +8,6 @@
 {% block profile-bar-profile %}active{% endblock %}
 
 {% block content %}
-  {% with profile.get_form as form %}
```
Comment (gregmuellegger) [inline readthedocs/templates/profiles/public/profile_detail.html]: A `{% with  %}` always has a `{% endwith %}` that should be removed as well ;)

**Pull request merged**